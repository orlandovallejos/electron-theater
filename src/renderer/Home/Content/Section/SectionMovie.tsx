/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import PosterItem from 'renderer/common/PosterItem';
// Types
import { Movie } from 'renderer/../types';
// Styles
import styles from './index.scss';

type Props = {
  movies: Movie[];
};

const SectionMovie = (props: Props) => {
  const { movies } = props;
  return (
    <div className={styles.wrapper}>
      {movies.map((item) => {
        return (
          <PosterItem
            key={item.id}
            url={item.poster_path}
            title={item.title}
            voteAverage={item.vote_average}
            releaseDate={item.release_date}
          />
        );
      })}
    </div>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default SectionMovie;
