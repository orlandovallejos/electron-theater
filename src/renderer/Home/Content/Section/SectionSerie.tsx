/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import PosterItem from '../../../common/PosterItem';
// Types
import { Serie } from '../../../../types';
// Styles
import SectionWrapper from './style';

type Props = {
  series: Serie[];
};

const SectionSerie = (props: Props) => {
  const { series } = props;
  return (
    <SectionWrapper>
      {series.map((item) => {
        return (
          <PosterItem
            key={item.id}
            url={item.poster_path}
            title={item.name}
            voteAverage={item.vote_average}
            releaseDate={item.first_air_date}
          />
        );
      })}
    </SectionWrapper>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default SectionSerie;
