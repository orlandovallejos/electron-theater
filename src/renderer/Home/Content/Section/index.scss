.wrapper {
  background-color: #101010;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
