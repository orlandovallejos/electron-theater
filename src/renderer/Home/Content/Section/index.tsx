/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import PosterItem from '../../../common/PosterItem';
// Types
import { MovieItem } from '../../../../types';
// Styles
import SectionWrapper from './style';

type Props = {
  items: MovieItem[];
};

const Section = (props: Props) => {
  const { items } = props;
  return (
    <SectionWrapper>
      {items.map((item) => {
        return (
          <PosterItem
            key={item.id}
            url={item.posterPath}
            title={item.title}
            voteAverage={item.voteAverage}
            releaseDate={item.releaseDate}
          />
        );
      })}
    </SectionWrapper>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Section;
