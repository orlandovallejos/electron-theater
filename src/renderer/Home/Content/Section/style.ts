/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import styled from 'styled-components';

const SectionWrapper = styled.div`
  background-color: #101010;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;
SectionWrapper.displayName = 'SectionWrapper';

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default SectionWrapper;
