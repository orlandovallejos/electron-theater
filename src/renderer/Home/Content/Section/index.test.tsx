/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import shallow from 'enzyme/shallow';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Section from './index';

describe('<Section />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Section
        items={[
          {
            id: 'item-01',
            backdropPath: 'backdrop path',
            posterPath: 'poster path',
            title: 'My movie',
            voteAverage: 7,
            releaseDate: '2021-04-19',
          },
        ]}
      />
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
