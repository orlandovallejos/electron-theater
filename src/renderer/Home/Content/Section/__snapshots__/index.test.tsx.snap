// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render correctly 1`] = `
"<SectionWrapper>
  <PosterItem id=\\"item-01\\" url=\\"poster path\\" title=\\"My movie\\" voteAverage={7} releaseDate=\\"2021-04-19\\" baseUrl={[undefined]} />
</SectionWrapper>"
`;
